activerecord/test/cases/query_cache_test.rb

```diff
@@ -300,7 +300,6 @@ def test_query_cache_across_threads
 
         thread_1_connection = ActiveRecord::Base.lease_connection
         ActiveRecord::Base.connection_handler.clear_active_connections!(:all)
-        assert_cache :off, thread_1_connection
```

Comment:
`clear_active_connections!` disables the query cache but the cache store is still dirty. We can't `assert_cache :off` anymore because this would return the pool that is dirty.

I'm not even sure if disabling the query cache when connections are released is the right behavior. Maybe the intent is to just clear the query cache? Because when you checkout a connection again the query cache would still be disabled for the thread.

In any case, this is a separate issue.